src/api.py

```diff
@@ -25,7 +27,32 @@ def home():
         return 'WELCOME TO CABINET'
 
 
-    @app.route('/blob', methods=['GET', 'POST', 'PUT'])
+    @app.route('/hosts', methods=['GET'])
+    def hosts():
+        with open('config/config.yaml','r') as file:
+            config_dict = yaml.safe_load(file) 
+        hosts = list(config_dict['save_hosts'].keys())
+        return Response(body={'hosts':hosts}).json() 
+        
+
+    @app.route('/paths', methods=['GET'])
+    def paths():
+        try:
+            conn, cur = db_connect(env=env)
+            new_blob_unsaved = PathsSchema(request.args.to_dict())
+            blob_type = new_blob_unsaved.blob_type
+            if not blob_type in BLOB_TYPES.keys():
+                return Response(status_code=400,error_message= f'InvalidBlobType: {blob_type} blob_type does not exist').json()
+            if f.duplicate(new_blob_unsaved.blob_hash, cur):
+                return Response(status_code=400, error_message='BlobDuplication: blob already in cabinet').json()
+            save_paths = f.generate_paths(new_blob_unsaved)
+            return Response(body={'paths':save_paths}).json()
+        finally:
+            cur.close()
+            conn.close()
+            
+
+    @app.route('/blob', methods=['GET', 'POST'])
     def blob(): 
         try:
             conn, cur = db_connect(env=env)
```

Comment:
move ```blob_type = user_search['blob_type']``` down.

Reply:
deleted variable assignment and just wrote path where needed